swap/src/cli/watcher.rs

```diff
@@ -45,6 +48,25 @@ impl Watcher {
         loop {
             tokio::time::sleep(Duration::from_secs(Watcher::CHECK_INTERVAL)).await;
 
+            // Fetch the current Bitcoin balance
+            match self.wallet.balance().await {
```

Comment:
FYI: this method uses the locally stored utxo set to calculate the balance. If a new transaction arrived since the last call to .sync(), it'll not be reflected in the returned balance.

I think it's fine to leave it like this for. In the future (once we add more wallet functionality) we can revisit this and subscribe to our wallet addresses.

Reply:
Yeah, I didn't want to force sync so often.